nvflare/private/fed/server/job_cmds.py

```diff
@@ -422,6 +435,39 @@ def get_job_meta(self, conn: Connection, args: List[str]):
                     f"job {job_id} does not exist", meta=make_meta(MetaStatusValue.INVALID_JOB_ID, job_id)
                 )
 
+    def list_job_components(self, conn: Connection, args: List[str]):
+        if len(args) < 2:
+            conn.append_error("Usage: list_job_components job_id", meta=make_meta(MetaStatusValue.SYNTAX_ERROR))
+            return
+
+        job_id = conn.get_prop(self.JOB_ID)
+        engine = conn.app_ctx
+        job_def_manager = engine.job_def_manager
+        if not isinstance(job_def_manager, JobDefManagerSpec):
+            raise TypeError(
+                f"job_def_manager in engine is not of type JobDefManagerSpec, but got {type(job_def_manager)}"
+            )
+        with engine.new_context() as fl_ctx:
+            list_of_data = job_def_manager.list_components(jid=job_id, fl_ctx=fl_ctx)
+            if list_of_data:
+                system_components = {"workspace", "meta", "scheduled", "data"}
+                filtered_data = [item for item in list_of_data if item not in system_components]
+                if filtered_data:
+                    data_str = ", ".join(filtered_data)
+                    conn.append_string(data_str)
+                    conn.append_success(
+                        "", meta=make_meta(MetaStatusValue.OK, extra={MetaKey.JOB_COMPONENTS: data_str})
```

Comment:
To make the meta content easier for programmers, don't use data_str here (otherwise the programmer will have to parse it). You can directly use filtered_data which is already a parsed list.